rust/lance-linalg/build.rs

```diff
@@ -14,6 +15,19 @@ fn main() {
             .flag("-Wpedantic")
             .compile("f16");
     }
+    if cfg!(all(target_os = "macos", feature = "FEAT_BF16")) {
```

Comment:
is `FEAT_BF16` a rustc feature? does this ever compiled?

Reply:
It's [mac](https://developer.apple.com/documentation/kernel/1387446-sysctlbyname/determining_instruction_set_characteristics#3915618) and I've not tested this on M2/M3 so I can't tell yet.

Reply:
How could we proceed with M2/M3 development? Shall I just buy a mac mini?

Reply:
It's also very likely arm/apple intrinsics have different names.

Reply:
oh, no. I meant that `FEAT_BF16` is a `sysctl -n | grep cpu` output.

It is not a `target-feature` recognized by the compiler.

Reply:
Yeah I only have an M1 so I didn't explore this further. Will remove the mac compilation and it can be a follow-up issue.